publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java

```diff
@@ -553,6 +555,20 @@ void shouldValidateOkIfInstanceTypeIsNotSet() throws IOException {
         assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_CREATED)));
     }
 
+    @Test
+    void shouldReturnBadRequestWhenAttemptingToPersistPublicationWithMultipleCristinIdentifiers() throws Exception {
+        var request = createEmptyPublicationRequest();
+        request.setAdditionalIdentifiers(Set.of(new AdditionalIdentifier("Cristin", randomString()),
+                                                new AdditionalIdentifier("Cristin", randomString())));
+        var inputStream = createPublicationRequest(request);
+        handler.handleRequest(inputStream, outputStream, context);
+
+        var actual = GatewayResponse.fromOutputStream(outputStream, Problem.class);
+        assertThat(actual.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
+        var publicationResponse = actual.getBodyObject(Problem.class);
+        assertThat(publicationResponse.getDetail(), containsString("Multiple Cristin identifiers are not allowed"));
```

Comment:
The fact we have more than one assertion here makes me wish we used a testing framework that supports rest testing — not your issue, but we should look into this.